sdk/js/src/Store.ts

```diff
@@ -16,53 +16,38 @@ export class Store {
         this.logger = logger
     }
 
-    save(storeKey: string, data: unknown): Promise<void> {
-        return this.modifyStore((storeKey: string) => {
-            this.store?.setItem(storeKey, JSON.stringify(data))
-        }, storeKey)
+    save(storeKey: string, data: unknown): void {
+        this.store?.setItem(storeKey, JSON.stringify(data))
     }
 
-    load(storeKey: string): Promise<string | null | undefined> {
-        return new Promise((resolve) => {
-            resolve(this.store?.getItem(storeKey))
-        })
+    load(storeKey: string): string | null | undefined {
+        return this.store?.getItem(storeKey)
     }
 
-    clear(storeKey: string): Promise<void> {
-        return this.modifyStore((storeKey: string) => this.store?.removeItem(storeKey), storeKey)
-    }
-
-    modifyStore(modify: (storeKey: string) => void, storeKey: string): Promise<void> {
-        try {
-            return new Promise((resolve) => {
-                resolve(modify(storeKey))
-            })
-        } catch (e) {
-            this.logger?.error(e as string)
-            return Promise.resolve()
-        }
+    remove(storeKey: string): void {
+        this.store?.removeItem(storeKey)
     }
 
     saveConfig(data: BucketedUserConfig): Promise<void> {
-        return this.save(StoreKey.Config, data)
+        return new Promise((resolve) => {
+            resolve(this.save(StoreKey.Config, data))
+        })
     }
 
-    loadConfig(): Promise<string | null | undefined> {
+    loadConfig(): string | null | undefined {
         return this.load(StoreKey.Config)
     }
 
     saveUser(user: DVCPopulatedUser): Promise<void> {
```

Comment:
this shouldn't be a promise either

Reply:
but in `Client.ts`, seems like we want to log the success message. If it's not a promise, we don't know if it's succeeded. Do we want to keep the logger and promise or just remove both? same for the `saveConfig` case
```
this.store.saveUser(user).then(() => {
       this.logger.info('Successfully saved user to local storage')
})
```

Reply:
you can just log it here instead of the client